staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -154,18 +154,52 @@ func evaluatePolicyAndCreateAuditEvent(req *http.Request, policy audit.PolicyRul
 	}, nil
 }
 
+// writeLatencytoAnnotation writes the latency incurred in different
+// layers of the apiserver to the annotations of the audit object.
+// it should be invoked after ev.StageTimestamp has been set appropriately.
+func writeLatencytoAnnotation(ctx context.Context, ev *auditinternal.Event) {
+	// we will track latency in annotation only when the total latency
+	// of the given request exceeds 500ms, this is in keeping with the
+	// traces in rest/handlers for create, delete, update, and get.
```

Comment:
What about list and deletecolection.  I would expect those to be excluded as well or at least have a different threshold.

Reply:
same, `500ms` is the threshold for list and deletecollection, i updated the comment.